python/pyiceberg/table/__init__.py

```diff
@@ -14,17 +14,102 @@
 # KIND, either express or implied.  See the License for the
 # specific language governing permissions and limitations
 # under the License.
-
-from typing import Optional, Union
+from functools import cached_property
+from typing import (
+    Dict,
+    List,
+    Optional,
+    Union,
+)
 
 from pydantic import Field
 
+from pyiceberg.schema import Schema
 from pyiceberg.table.metadata import TableMetadataV1, TableMetadataV2
+from pyiceberg.table.partitioning import PartitionSpec
+from pyiceberg.table.snapshots import Snapshot, SnapshotLogEntry
+from pyiceberg.table.sorting import SortOrder
 from pyiceberg.typedef import Identifier
 from pyiceberg.utils.iceberg_base_model import IcebergBaseModel
 
 
 class Table(IcebergBaseModel):
     identifier: Identifier = Field()
     metadata_location: str = Field()
-    metadata: Optional[Union[TableMetadataV1, TableMetadataV2]] = Field()
+    metadata: Union[TableMetadataV1, TableMetadataV2] = Field()
+
+    def refresh(self, catalog) -> "Table":
```

Comment:
Let's throw an exception and discuss this in a separate PR.

Reply:
Reverted 👍🏻